packages/modules/devices/sungrow/sungrow/bat.py

```diff
@@ -33,52 +31,70 @@ def initialize(self) -> None:
         self.store = get_bat_value_store(self.component_config.id)
         self.fault_state = FaultState(ComponentInfo.from_component_config(self.component_config))
         self.last_mode = 'Undefined'
-        self.firmware_check = self.check_firmware_register()
+        self.register_check = self.detect_register_check()
 
-    def check_firmware_register(self) -> bool:
-        if Firmware(self.device_config.configuration.firmware) == Firmware.v1:
-            return False
+    def detect_register_check(self) -> str:
+        # Battery register availability test
         unit = self.device_config.configuration.modbus_id
+
         try:
             self.__tcp_client.read_input_registers(5213, ModbusDataType.INT_32,
                                                    wordorder=Endian.Little, unit=unit)
-            log.debug("Wechselrichter Firmware ist größer gleich 95.09")
-            return True
+            self.__tcp_client.read_input_registers(5630, ModbusDataType.INT_16, unit=unit)
+            log.debug("Battery register check: using new_registers (5213/5630).")
+            return 'new_registers'
```

Comment:
```suggestion
            return RegMode.NEW_REGISTERS
```
Kannst Du für den Modus bitte ein Enum verwenden?